### DISCUSSION TITLE
FileDropWidget

### DISCUSSION THREAD

**Discussion by codematsing** [jrief/django-formset#89] [Ideas]
Hoping this library could also support or integrate Dropzone widget in the future! :)

Similar to this:
https://github.com/pedrodcb/django-dropzone

**Comment by jrief** [comment]
You mean something like this: https://django-formset.fly.dev/uploading/

Which features Dropzone offers, are missing in django-formset?

**Comment by codematsing** [comment]
I like how files are handled asynchronously and the drag and drop feature of this widget. I especially like how this handles FileFields between CRUD Views which used to be a pain in vanilla Django... But I'm looking for solution that particularly handles multiple files in drop zone. While there are other libraries, I do like how your library tries to encapsulate different usecases and tries to be the standard for Form handling. It would just be cool to have it handle this usecase too.

A more concrete example could be a relationship such as a `Folder` holding multiple `Files` (like Google Drive)

I'm sure form collection could be the workaround but it would mean having to manual add a field for every new file.

If there could be a dropzone that could handle multiple files and have the developer handle the saving functionality in the View - similar to FormCollectionView, that would be great!

**Comment by jrief** [comment]
> But I'm looking for solution that particularly handles multiple files in drop zone.

Django's [FileField](https://docs.djangoproject.com/en/4.2/ref/forms/fields/#django.forms.FileField) can only handle one file at a time. Therefore, although easily possible, I did not implement this in the upload widget. The next question which would arise then, is where to keep those multiple file paths.

> I'm sure form collection could be the workaround but it would mean having to manual add a field for every new file.

This is how I handle it in my real-world projects.

> If there could be a dropzone that could handle multiple files and have the developer handle the saving functionality in the View - similar to FormCollectionView, that would be great!

I will think about this. However, then the user of this library has to do quite some transformations when converting his models into forms and vice versa. I believe that the FormCollection approach then is simpler.
